packages/medusa/src/services/claim.ts

```diff
@@ -262,76 +367,18 @@ export default class ClaimService extends TransactionBaseService {
           addressId = saved.id
         }
 
-        if (type !== ClaimType.REFUND && type !== ClaimType.REPLACE) {
-          throw new MedusaError(
-            MedusaError.Types.INVALID_DATA,
-            `Claim type must be one of "refund" or "replace".`
-          )
-        }
-
-        if (type === ClaimType.REPLACE && !additional_items?.length) {
-          throw new MedusaError(
-            MedusaError.Types.INVALID_DATA,
-            `Claims with type "replace" must have at least one additional item.`
-          )
-        }
-
-        if (!claim_items?.length) {
-          throw new MedusaError(
-            MedusaError.Types.INVALID_DATA,
-            `Claims must have at least one claim item.`
-          )
-        }
-
-        if (refund_amount && type !== ClaimType.REFUND) {
-          throw new MedusaError(
-            MedusaError.Types.INVALID_DATA,
-            `Claim has type "${type}" but must be type "refund" to have a refund_amount.`
-          )
-        }
-
         let toRefund = refund_amount
         if (type === ClaimType.REFUND && typeof refund_amount === "undefined") {
-          const lines = claim_items.map((ci) => {
-            const allOrderItems = order.items
-
-            if (order.swaps?.length) {
-              for (const swap of order.swaps) {
-                swap.additional_items.forEach((it) => {
-                  if (
-                    it.shipped_quantity ||
-                    it.shipped_quantity === it.fulfilled_quantity
```

Comment:
comment: @srindom Would be great with an additional pair of eyes on this check, as it significantly changed.

In the existing implementation, we never check the second part of the condition, as `shipped_quantity` would be true for any given number - regardless of how many were actually shipped or fulfilled